.github/workflows/github-actions.yml

```diff
@@ -0,0 +1,40 @@
+# loosely based on https://github.com/actions/starter-workflows/blob/main/ci/python-app.yml, which is MIT Licensed
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: temurin-vdr-generator ci
+
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
```

Comment:
please pin these to the actual sha

Reply:
e.g `actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f #v4.1.3`

Reply:
where can i find the actual shas?